scripts/ci-entrypoint.sh

```diff
@@ -139,6 +139,10 @@ install_addons() {
     # export the target cluster KUBECONFIG if not already set
     export KUBECONFIG="${KUBECONFIG:-${PWD}/kubeconfig}"
 
+    # wait for the apiserver pod to be Ready.
+    APISERVER_POD=$(${KUBECTL} get pods -n kube-system -o name | grep apiserver)
+    "${KUBECTL}" wait --for=condition=Ready -n kube-system "${APISERVER_POD}" --timeout=5m
+
```

Comment:
Thinking if a kubectl describe of pod will help in useful logs after the timeout.

Reply:
that's already part of the cleanup step https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3091/files#diff-2009f7be5363812e22862771f557363795759e67b3ca2e32c50d9fd5f8c5669bR260

Reply:
actually I lied, the logger collects events and pod logs, but trying to see if something like [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3097) could work to print a full kubectl describe pod in artifacts instead of just events

Reply:
yep that works, see #3097